src/components/ec-currency-input/ec-currency-input.vue

```diff
@@ -0,0 +1,176 @@
+<template>
+  <div class="ec-currency-input">
+
+    <label
+      v-if="label || note"
+      class="ec-currency-input__label"
+      :for="id"
+    >
+      <span
+        v-if="label"
+        data-test="ec-currency-input__label-text"
+        class="ec-currency-input__label-text"
+      >{{ label }}</span>
+
+      <span
+        v-if="note"
+        data-test="ec-currency-input__note"
+        class="ec-currency-input__note"
+      >{{ note }}</span>
+    </label>
+
+    <div
+      ref="popperWidthReference"
+      class="ec-currency-input__input-group"
+    >
+      <ec-dropdown
+        v-model="inputModel.currency"
+        :class="{'ec-currency-input__currencies--on-focus': currenciesOnFocus}"
+        :id-from-parent="id"
+        :items="currencies"
```

Comment:
I would expect passing array of currencies `['GBP', 'EUR']` via props and then this component to convert it into object that ec-dropdown can understand. Then, when you select new value in the dropdown you should update the model with currency value and not with entire item.